unix/vncserver/vncsession.c

```diff
@@ -360,6 +365,18 @@ redir_stdio(const char *homedir, const char *display)
             syslog(LOG_CRIT, "Failure creating \"%s\": %s", logfile, strerror(errno));
             _exit(EX_OSERR);
         }
+
+#ifdef HAVE_SELINUX
+        int result;
+        if (selinux_file_context_verify(logfile, 0) == 0) {
+            result = selinux_restorecon(logfile, SELINUX_RESTORECON_RECURSE);
+
+            if (result < 0) {
+                syslog(LOG_CRIT, "Failure restoring SELinux context for \"%s\"", logfile);
```

Comment:
It might be useful to also log the error reported by restorecon `strerror(errno)`.

Reply:
Updated. Thanks.